kernel/src/transaction.rs

```diff
@@ -0,0 +1,259 @@
+use std::iter;
+use std::sync::Arc;
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use crate::actions::get_log_schema;
+use crate::error::Error;
+use crate::expressions::{Scalar, StructData};
+use crate::path::ParsedLogPath;
+use crate::schema::{MapType, StructField, StructType};
+use crate::snapshot::Snapshot;
+use crate::{DataType, DeltaResult, Engine, EngineData, Expression, Version};
+
+const KERNEL_VERSION: &str = env!("CARGO_PKG_VERSION");
+const UNKNOWN_OPERATION: &str = "UNKNOWN";
+
+/// A transaction represents an in-progress write to a table.
+pub struct Transaction {
```

Comment:
I think (as @scottsand-db brought up) likely in the near future we want to move to `TransactionBuilder` to force the engine to specify required fields (for now just commit info - we can think about making operation required, for now we just set operation to `UNKNOWN` if it is omitted). Flagging here in case people feel strongly we should do now

Reply:
How is commit info a required field, if it's passed as `Option`?

Reply:
Yup I think this is just a failure of my API. we could make it required here and then need it passed to `Transaction::new(..)` but see other comment for if we just want to go ahead and move to builder pattern for this and operation